contracts/SwapAndCall.sol

```diff
@@ -0,0 +1,80 @@
+pragma solidity ^0.5.0;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import 'hardlydifficult-ethereum-contracts/contracts/proxies/CallContract.sol';
+
+
+/**
+ * @notice call any contract(s) and then refund the remaining balance (if any)
+ * Idea from https://etherscan.io/address/0x11111254369792b2Ca5d084aB5eEA397cA8fa48B#code
+ */
+contract SwapAndCall
+{
+  using CallContract for address;
+  using SafeERC20 for IERC20;
+
+  // Accept ETH from other contracts
+  function() external payable {}
+
+  /**
+   * @notice Make a series of calls and then refunds any funds remaining.
+   * @param _contractsAndTokensToRefund The contract address for each call to make.
+   * Followed by the address of any ERC-20 tokens to refund at the end.
+   * ETH is always refunded.  If no additional token is required this may be empty.
+   * @param _callDataConcat Call data for each call, appended into a single bytes field.
+   * @param _startPositionsAndValues The start position of callData for each call after
+   * the first (assumed to be 0).
+   * Followed by the call value to include with each call.
+   * @dev Variables are reused to avoid a `stack too deep` error.
```

Comment:
Can you please explain this a little more? by reused are you referring to how `_contractsAndTokensToRefund` is used for both the addresses of the contracts to call, AND erc-20 contract addresses?

Reply:
Yea, that and the code reuses variables.

Solidity limits the stack to just 16.  So without this I had hit the dreaded stack too deep.